File: qc-application/src/main/java/com/eager/questioncloud/application/exception/PaymentExceptionHandler.java
package com.eager.questioncloud.application.exception;

import com.eager.questioncloud.core.exception.InvalidPaymentException;
import com.eager.questioncloud.pg.exception.InvalidPaymentIdException;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ControllerAdvice;
import org.springframework.web.bind.annotation.ExceptionHandler;

@ControllerAdvice
@RequiredArgsConstructor
public class PaymentExceptionHandler {
    @ExceptionHandler(InvalidPaymentException.class)
    protected ResponseEntity<ErrorResponse> handleInvalidPaymentException() {
        return ResponseEntity
            .status(HttpStatus.BAD_REQUEST)
            .body(
                ErrorResponse
                    .builder()
                    .status(400)
                    .message("결제 금액 오류")
                    .build()
            );
    }

    @ExceptionHandler(InvalidPaymentIdException.class)
    protected ResponseEntity<ErrorResponse> handleInvalidPaymentIdException() {
        return ResponseEntity
            .status(HttpStatus.BAD_REQUEST)
            .body(
                ErrorResponse
                    .builder()
                    .status(404)
                    .message("존재하지 않는 결제입니다.")
                    .build()
            );
    }
}

File: qc-application/src/main/kotlin/com/eager/questioncloud/application/exception/PaymentExceptionHandler.kt
package com.eager.questioncloud.application.exception

import com.eager.questioncloud.core.exception.InvalidPaymentException
import com.eager.questioncloud.pg.exception.InvalidPaymentIdException
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.ControllerAdvice
import org.springframework.web.bind.annotation.ExceptionHandler

@ControllerAdvice
class PaymentExceptionHandler {
    @ExceptionHandler(InvalidPaymentException::class)
    protected fun handleInvalidPaymentException(): ResponseEntity<ErrorResponse> {
        return ResponseEntity
            .status(HttpStatus.BAD_REQUEST)
            .body(ErrorResponse(400, "결제 금액 오류"))
    }

    @ExceptionHandler(InvalidPaymentIdException::class)
    protected fun handleInvalidPaymentIdException(): ResponseEntity<ErrorResponse> {
        return ResponseEntity
            .status(HttpStatus.BAD_REQUEST)
            .body(ErrorResponse(404, "존재하지 않는 결제입니다."))
    }
}
